File: Samples/Forms and Annotations/FindControlByName/C#/FindControlByName.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class FindControlByName
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            const string pathToFile = "FindControlByName.pdf";

            using (var pdf = new PdfDocument(@"..\Sample Data\form.pdf"))
            {
                if (pdf.TryGetControl("email", out var control) &&
                    control is PdfTextBox emailTextBox)
                {
                    emailTextBox.Text = "support@bitmiracle.com";
                }

                pdf.Save(pathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pathToFile) { UseShellExecute = true });
        }
    }
}

File: Samples/Forms and Annotations/FindControlByName/VB.NET/FindControlByName.vb
Imports BitMiracle.Docotic
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class FindControlByName
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Const pathToFile As String = "FindControlByName.pdf"

            Using pdf As New PdfDocument("..\Sample Data\form.pdf")
                Dim emailTextBox As PdfTextBox = GetTextBox(pdf, "email")
                If emailTextBox IsNot Nothing Then
                    emailTextBox.Text = "support@bitmiracle.com"
                End If

                pdf.Save(pathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(pathToFile) With {.UseShellExecute = True})
        End Sub

        Private Shared Function GetTextBox(pdf As PdfDocument, name As String) As PdfTextBox
            Dim control As PdfControl = Nothing
            If Not pdf.TryGetControl(name, control) Then
                Return Nothing
            End If

            Return TryCast(control, PdfTextBox)
        End Function
    End Class
End Namespace